CIP-0118/README-implicit.md

```diff
@@ -0,0 +1,937 @@
+users for the details of their usecases (see [Callout to the Community](#callout-to-the-community))
+
+## Attacks and Mitigations
+
+### DDoS
+
+**Off-Chain**.
+Since we do not propose an off-chain component, we do not discuss mitigation of
+an off-chain DDoS attack, however, we note that such an attack consists of
+communicating intents that may never be fulfilled. This is because either they
+offer undesirable tokens for swaps, or set prices too high. It will be difficult
+to determine whether any intent _will_ ever be fulfilled, or even whether it was
+created with a malicious intent (spam).
+
+**On-Chain (Mempool)**.
+The possibility of un-fulfillable intents is an important reason for
+transmitting only entire zones across the Cardano network. Upon validating an
+incoming zone, a node is able to
+determine whether to discard it or add it to the mempool. This is not the
+case for intent-containing transactions. If we were to allow the mempool to
+contain intent-containing transactions outside zones, such a transaction
+could be placed in a mempool indefinitely, waiting for an appropriate
+partially-valid counterpart to complete the zone. An attacker could fill up
+a node's mempool with such transactions, rendering it unable to record
+additional incoming transactions. Zone-based validation
+prevents this attack.
+
+**On-Chain (Phase-1)**.
+Since zones are size-restricted by the constraint that currently restricts the
+size of a single transaction (`maxTxSize`), we assume that the phase-1 validation cost
+has the same limitations as before. So, phase-1 validation of a single
+zone is likely to cost the same as phase-1 validation of a single transaction
+currently costs. Therefore, we conjecture that the risks of a spam attack using phase-1
+invalid entities is not increased as compared to what it currently is.
+
+**On-Chain (Phase-2)**. The mempool attack scenario describes mitigation of an attack
+which results in a DDoS by using up all the node's mempool storage. Another kind of
+DDoS attack which we must defend against is one that involves a malicious
+party sending transactions with large (i.e. memory- or CPU-use-intensive)
+failing scripts. We refer to a script failure in a transaction as a _phase-2_ failure.
+
+In the current ledger design, this is mitigated by requiring a transaction to provide
+an amount of Ada collateral that is proportional to the sum of the sizes of all
+its scripts. This collateral is collected in the case of script failure.
+A phase-2 transaction failure in the design proposed in this CIP may result in phase-1 validation
+failures of subsequent transactions.
+To address this problem, we have defined the `ZONE-N` rule of `ZONE`. It requires
+that the collateral of each transaction
+
+- comes from the UTxO set to which the _zone_, rather than the transaction, is applied
+- covers the size-based fee for the transactions corresponding to `requiredTxs` transactions 
+in addition to itself
```

Comment:
requiredTx does not guarantee fairness either.

Say this is the zone: 
``
[ tx1a <- tx2a <- tx3a
, tx1b
, tx1c <- tx2c
]
```
`a`, `b` and `c` groups provide their own collateral for the group. What happens to the whole zone when only `tx1b` fails phase2 validation?

Collateral for group `a` is not gonna be collected, but the node did the work validating that dependency group.

Reply:
First, you do phase-1 checks, which includes checking that exactly 0 or 1 transactions in a zone are labeleld as `isValid = False`. Then 

- if no transactions are `isValid` False, all scripts should succeed (unless its a bad block and that's bad)
- if 1 tx has `isValid` False, you only run the scripts in that transaction, not in any of its dependents. You just pass the dependent transactions as `TxInfo` to those scripts (to check that they fail)

Reply:
Same comment. isValid is set by cardano-node, not by the people building the transaction